src/Fieldtypes/ProductQueryBuilder.php

```diff
@@ -0,0 +1,303 @@
+        }
+
+        $globalMethod = strtolower($data['globalConjunction'] ?? 'AND') === 'and' ? 'where' : 'orWhere';
+
+        foreach ($data['groups'] as $group) {
+            if (empty($group['conditions'])) {
+                continue;
+            }
+
+            $query->$globalMethod(function($query) use ($group, $table) {
+                foreach ($group['conditions'] as $condition) {
+                    if (empty($condition['attribute']) || empty($condition['operator'])) {
+                        continue;
+                    }
+
+                    $value = $condition['value'];
+                    $operator = $condition['operator'];
+                    $column = $table . '.' . $condition['attribute'];
+                    $method = strtolower($group['conjunction'] ?? 'AND') === 'and' ? 'where' : 'orWhere';
+
+                    $query = $this->applyOperator($query, $method, $column, $operator, $value);
+                }
+            });
+        }
+
+        return $query->limit($data['limit'] ?? 100)->pluck($table . '.sku')->toArray();
+    }
+
+    public function toSkusEav($data): array
+    {
+        $storeId = config('rapidez.store');
+
+        $innerQuery = DB::table('catalog_product_entity')
+            ->select('catalog_product_entity.entity_id', 'catalog_product_entity.sku');
+
+        $attributeJoins = [];
+        foreach ($data['groups'] as $group) {
+            if (empty($group['conditions'])) {
+                continue;
+            }
+
+            foreach ($group['conditions'] as $i => $condition) {
+                if (empty($condition['attribute']) || empty($condition['operator'])) {
+                    continue;
+                }
+
+                $attribute = $condition['attribute'];
+                if (!isset($attributeJoins[$attribute])) {
+                    $backendType = DB::table('eav_attribute')
+                        ->join('eav_entity_type', 'eav_entity_type.entity_type_id', '=', 'eav_attribute.entity_type_id')
+                        ->where('eav_entity_type.entity_type_code', '=', 'catalog_product')
+                        ->where('eav_attribute.attribute_code', '=', $attribute)
+                        ->value('backend_type');
+
+                    if ($backendType && $backendType !== 'static') {
+                        $attributeJoins[$attribute] = [
+                            'type' => $backendType,
+                            'instances' => []
+                        ];
+                    }
+                }
+
+                if (isset($attributeJoins[$attribute])) {
+                    $attributeJoins[$attribute]['instances'][] = $i;
+                }
+            }
+        }
+
+        foreach ($attributeJoins as $attribute => $info) {
+            foreach ($info['instances'] as $i) {
+                $alias = "eav_{$attribute}_{$i}";
+                $backendType = $info['type'];
+                $attributeId = DB::table('eav_attribute')
```

Comment:
There is a Attribute model https://github.com/rapidez/core/blob/3f97927fbc149ba8e472de32255f51cd88d2ec9f/src/Models/Attribute.php#L10